packages/amplify-provider-awscloudformation/lib/build-resources.js

```diff
@@ -19,14 +19,17 @@ async function buildResource(context, resource) {
   const backEndDir = context.amplify.pathManager.getBackendDirPath();
   const projectRoot = context.amplify.pathManager.searchProjectRootPath();
   const resourceDir = path.normalize(path.join(backEndDir, category, resourceName, 'src'));
-  const packageJsonPath = path.normalize(path.join(backEndDir, category, resourceName, 'src', 'package.json'));
-  const packageJsonMeta = fs.statSync(packageJsonPath);
+  const lastUpdateTime = fs.readdirSync(resourceDir)
+    .map(file => path.join(resourceDir, file))
+    .map(file => fs.statSync(file))
+    .map(stats => stats.mtime)
+    .reduce((acc, it) => it > acc ? it : acc, 0); // reduce to the latest time
   const distDir = path.normalize(path.join(backEndDir, category, resourceName, 'dist'));
 
   let zipFilename = resource.distZipFilename;
   let zipFilePath = zipFilename ? path.normalize(path.join(distDir, 'latest-build.zip')) : '';
 
-  if (!resource.lastBuildTimeStamp || new Date(packageJsonMeta.mtime) > new Date(resource.lastBuildTimeStamp)) {
+  if (!resource.lastBuildTimeStamp || new Date(lastUpdateTime) > new Date(resource.lastBuildTimeStamp)) {
```

Comment:
This is a runtime provider concern.

Reply:
We could punt this to the runtime provider. However, I think it's safe to say agnostic of runtime that if a file has changed in the source folder that you need to rebuild the function. Then the runtime provider can assume that if they get a request to build, it means that the function actually needs to be rebuilt.

Even if we decide to move this into the runtime provider, I think this change should stay because it will rebuild existing js functions if other files besides package.json have changed.